--- xmlsec-ltdl.c.orig	2003-09-08 19:21:17.000000000 -0700
+++ xmlsec-ltdl.c	2003-09-08 19:22:02.000000000 -0700
@@ -3234,31 +3230,31 @@
       /* If a specific path was passed, search only the directories
 	 listed in it.  */
       is_done = foreach_dirinpath (search_path, 0,
-				   foreachfile_callback, func, data);
+				   (xmlsec_lt_ptr)foreachfile_callback, func, data);
     }
   else
     {
       /* Otherwise search the default paths.  */
       is_done = foreach_dirinpath (user_search_path, 0,
-				   foreachfile_callback, func, data);
+				   (xmlsec_lt_ptr)foreachfile_callback, func, data);
       if (!is_done)
 	{
 	  is_done = foreach_dirinpath (getenv("LTDL_LIBRARY_PATH"), 0,
-				       foreachfile_callback, func, data);
+				       (xmlsec_lt_ptr)foreachfile_callback, func, data);
 	}
 
 #ifdef LTDL_SHLIBPATH_VAR
       if (!is_done)
 	{
 	  is_done = foreach_dirinpath (getenv(LTDL_SHLIBPATH_VAR), 0,
-				       foreachfile_callback, func, data);
+				       (xmlsec_lt_ptr)foreachfile_callback, func, data);
 	}
 #endif
 #ifdef LTDL_SYSSEARCHPATH
       if (!is_done)
 	{
 	  is_done = foreach_dirinpath (getenv(LTDL_SYSSEARCHPATH), 0,
-				       foreachfile_callback, func, data);
+				       (xmlsec_lt_ptr)foreachfile_callback, func, data);
 	}
 #endif
     }
